src/GraphQL/Execution/IExecutionContext.cs

```diff
@@ -78,6 +78,16 @@ public interface IExecutionContext : IProvideUserContext
         /// </summary>
         Variables Variables { get; }
 
+        /// <summary>
+        /// A dictionary of field argument values.
+        /// </summary>
+        IDictionary<GraphQLField, IDictionary<string, ArgumentValue>>? ArgumentValues { get; }
+
+        /// <summary>
+        /// A dictionary of directive argument values.
+        /// </summary>
+        IDictionary<GraphQLField, IDictionary<string, DirectiveInfo>>? DirectiveValues { get; }
```

Comment:
As of now, we provide means to retrieve directive from fields only, via `IResolveFieldContext`.  It would be an easy matter to change this dictionary to be ASTNode instead of GraphQLField and collective directives from anywhere in the AST, but (a) what's the point? and (b) where would the user access them from?

Reply:
I would not be puzzled without obvious need.